### PR TITLE
Introduces `coerce` to StringUtil

### PR DESCRIPTION
**Pull request by KaiVolland**
This introduces a `coerce` methode to the StringUtil.

@terrestris/devs plz review

### CODE REVIEW

File: src/Util/StringUtil/StringUtil.js [hunk 1]
```diff
@@ -14,6 +14,39 @@ class StringUtil {
 
     return text.replace(urlRegex, `<a href="$1" target="_blank">$1</a>`);
   }
+
+  /**
+   * This coereces the value of a string by casting it to the most plausible
```
Comment (marcjansen) [inline src/Util/StringUtil/StringUtil.js]: typo `coereces`

File: src/Util/StringUtil/StringUtil.js [hunk 2]
```diff
@@ -14,6 +14,39 @@ class StringUtil {
...
+  static coerce(string) {
+    if (!(string instanceof String || typeof string === 'string')) {
+      return string;
+    }
+
+    if (string.toLowerCase() === 'true') {
+      return true;
+    } else if (string.toLowerCase() === 'false') {
+      return false;
+    } else if (/^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/.test(string)) {
```
Comment (marcjansen) [inline src/Util/StringUtil/StringUtil.js]: This regex can be factored out and given a name

File: src/Util/StringUtil/StringUtil.js [hunk 3]
```diff
@@ -14,6 +14,39 @@ class StringUtil {
...
+      return parseFloat(string);
+    } else if (string.startsWith('[') && !string.startsWith('[[')) {
```
Comment (marcjansen) [inline src/Util/StringUtil/StringUtil.js]: Why wouldn't we coerce `[[123,345]]`?

File: src/Util/StringUtil/StringUtil.js [hunk 4]
```diff
@@ -14,6 +14,39 @@ class StringUtil {
...
+      return JSON.parse(string).map(a => StringUtil.coerce(a));
+    } else if (string.startsWith('{') && !string.startsWith('{{')) {
```
Comment (marcjansen) [inline src/Util/StringUtil/StringUtil.js]: Same question here.

File: src/Util/StringUtil/StringUtil.js [hunk 5]
```diff
@@ -14,6 +14,39 @@ class StringUtil {
 
     return text.replace(urlRegex, `<a href="$1" target="_blank">$1</a>`);
   }
+
+  /**
+   * This coereces the value of a string by casting it to the most plausible
+   * datatype, guessed by the value itself.
+   *
+   * @param {String} string The input string to coerce.
+   * @return {string|boolean|float|object} The coerced value.
```
Comment (marcjansen) [inline src/Util/StringUtil/StringUtil.js]: Missing array. It might make sense to simply say `*` (and probably more correct as well)

**Comment by dnlkoch** [comment]
Ping @KaiVolland.

File: src/Util/StringUtil/StringUtil.js (hunk 3 above)
Comment (KaiVolland) [reply src/Util/StringUtil/StringUtil.js]: I think this was some special BfS stuff. I'll remove it.

File: src/Util/StringUtil/StringUtil.js (hunk 4 above)
Comment (KaiVolland) [reply src/Util/StringUtil/StringUtil.js]: I think this was some special BfS stuff. I'll remove it.

**Comment by KaiVolland** [comment]
Any further comments or is this good to merge?

**Pull request merged**